libraries/botbuilder-dialogs-adaptive/src/actions/switchCondition.ts

```diff
@@ -66,8 +65,44 @@ export class SwitchCondition<O extends object = {}> extends Dialog<O> implements
             this._caseExpresssions = {};
             for (let i = 0; i < this.cases.length; i++) {
                 const caseScope = this.cases[i];
-                const caseCondition = Expression.equalsExpression(this.condition, caseScope.createValueExpression());
-                this._caseExpresssions[caseScope.value] = caseCondition;
+                const intVal = parseInt(caseScope.value);
+                if (!isNaN(intVal)) {
+                    // you don't have to put quotes around numbers, "23" => 23 OR "23".
```

Comment:
This whole thing is a bad idea.  I don't believe we should be loose in this way.  A number is a number and a string is a string.  What would I do if I wanted a case for each.